spec/std/bit_array_spec.cr

```diff
@@ -92,4 +92,37 @@ describe "BitArray" do
     ary = BitArray.new(64, true)
     ary.size.times { |i| ary[i].should be_true }
   end
+
+  it "reads bits from slice" do
+    ary = BitArray.new(43) # 5 bytes 3 bits
+    # 11010000_00000000_00001011_00000000_00000000_101xxxxx
+    ary[0] = true
+    ary[1] = true
+    ary[3] = true
+    ary[20] = true
+    ary[22] = true
+    ary[23] = true
+    ary[40] = true
+    ary[42] = true
+    slice = ary.to_slice
+
+    slice.size.should eq(6)
+    slice[0].should eq(0b00001011_u8)
+    slice[1].should eq(0b00000000_u8)
+    slice[2].should eq(0b11010000_u8)
+    slice[3].should eq(0b00000000_u8)
+    slice[4].should eq(0b00000000_u8)
+    slice[5].should eq(0b00000101_u8)
+  end
+
+  it "read bits written from slice" do
+    ary = BitArray.new(43) # 5 bytes 3 bits
+    slice = ary.to_slice
+    slice[0] = 0b10101010_u8
+    slice[1] = 0b01010101_u8
+    slice[5] = 0b11111101_u8
+    ary.each_with_index do |e, i|
+      e.should eq([1, 3, 5, 7, 8, 10, 12, 14, 40, 42].includes?(i))
```

Comment:
Maybe use a tuple here? It's a spec so performance doesn't matter much, but it's better if we also try to make specs run as fast as possible.